libraries/protocol/include/scorum/protocol/betting/wincase.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+#include <fc/static_variant.hpp>
+#include <scorum/protocol/betting/market_kind.hpp>
+#include <scorum/protocol/betting/threshold.hpp>
+
+namespace scorum {
+namespace protocol {
+namespace betting {
+
+struct home_tag;
+struct away_tag;
+struct draw_tag;
+struct both_tag;
+
+template <market_kind kind, typename tag = void> struct under;
+template <market_kind kind, typename tag = void> struct over
+{
+    static constexpr market_kind kind_v = kind;
+    using opposite_type = under<kind, tag>;
+
+    threshold_type threshold = 0;
+
+    opposite_type create_opposite() const
+    {
+        return opposite_type{ threshold };
+    }
+};
+
+template <market_kind kind, typename tag> struct under
+{
+    static constexpr market_kind kind_v = kind;
+    using opposite_type = over<kind, tag>;
+
+    threshold_type threshold = 0;
```

Comment:
Not good architecture when fields are duplicated in initial and opposite wincase.
Maybe try additional bool tag. For instance:

```
template <bool site, market_kind kind, typename tag = void> struct over_under
{
    static constexpr market_kind kind_v = kind;
    using opposite_type = over_under<!site, kind, tag>;

    threshold_type threshold = 0;

    opposite_type create_opposite() const
    {
        return opposite_type{ threshold };
    }
};
template <market_kind kind, typename tag = void> using over = over_under<true, kind, tag>;
template <market_kind kind, typename tag = void> using under = over_under<false, kind, tag>;
```